refinery/data_set_manager/views.py

```diff
@@ -235,15 +235,30 @@ def post(self, request, *args, **kwargs):
                 response = self.import_by_url(url)
             else:
                 response = self.import_by_file(f)
+
+            # get AWS Cognito identity ID
+            if settings.REFINERY_DEPLOYMENT_PLATFORM == 'aws':
+                try:
+                    identity_id = request.POST.get('identity_id')
+                except (KeyError, ValueError):
+                    error_msg = 'identity_id is missing'
+                    error = {'error_message': error_msg}
```

Comment:
True :) Originally, I've even had this simplified to return the same response for AJAX and non-AJAX requests but then decided to leave it as is since it's a pattern that's used in a few other places.

Reply:
It would be great to standardize error reporting API. I've added [api_error_response()](https://github.com/refinery-platform/refinery-platform/blob/69e894f9de1138898912473ad4f21d8358337619/refinery/core/utils.py#L985) utility function for that purpose.